x/validator-preference/msg_server.go

```diff
@@ -0,0 +1,68 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/osmosis-labs/osmosis/v12/x/validator-preference/types"
+)
+
+type msgServer struct {
+	keeper *Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper *Keeper) types.MsgServer {
+	return &msgServer{
+		keeper: keeper,
+	}
+}
+
+var _ types.MsgServer = msgServer{}
+
+func (server msgServer) SetValidatorSetPreference(goCtx context.Context, msg *types.MsgSetValidatorSetPreference) (*types.MsgSetValidatorSetPreferenceResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	preferences := msg.Preferences
+
+	// check if a user already have a validator-set created
+	existingValidators, found := server.keeper.GetValidatorSetPreference(ctx, msg.Delegator)
+
+	if found {
+		// check if the new preferences is the same as the existing preferences
+		if reflect.DeepEqual(preferences, existingValidators.Preferences) {
+			return nil, fmt.Errorf("The preferences are the same")
+		}
+
+		preferences = existingValidators.Preferences
+	}
+
+	// check if the distribution weights equals 1
+	err := server.keeper.ValidatePreferences(ctx, preferences)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Yeah I have a feeling if the user had a pre-existing validator preference set and then provided a preferences set that did not equal 1, this would not catch that issue due to problem above.

Reply:
Yup will fix this